subsys/net/openthread/rpc/server/ot_rpc_srp_client.c

```diff
@@ -0,0 +1,812 @@
+
+	nrf_rpc_rsp_send_uint(group, lease_int);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_srp_client_get_lease_interval,
+			 OT_RPC_CMD_SRP_CLIENT_GET_LEASE_INTERVAL,
+			 ot_rpc_cmd_srp_client_get_lease_interval,
+			 NULL);
+
+static void ot_rpc_cmd_srp_client_get_key_lease_interval(const struct nrf_rpc_group *group,
+							 struct nrf_rpc_cbor_ctx *ctx,
+							 void *handler_data)
+{
+	uint32_t key_lease_int;
+
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	key_lease_int = otSrpClientGetKeyLeaseInterval(openthread_get_default_instance());
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_uint(group, key_lease_int);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_srp_client_get_key_lease_interval,
+			 OT_RPC_CMD_SRP_CLIENT_GET_KEY_LEASE_INTERVAL,
+			 ot_rpc_cmd_srp_client_get_key_lease_interval, NULL);
+
+
+static void ot_rpc_cmd_srp_client_set_host_addresses(const struct nrf_rpc_group *group,
+						     struct nrf_rpc_cbor_ctx *ctx,
+						     void *handler_data)
+{
+	otError error;
+	size_t index = 0;
+
+	zcbor_list_start_decode(ctx->zs);
+
+	/* Check if more memory is needed */
+	if (host_data.addresses && (ctx->zs->elem_count > host_data.addresses_num)) {
+		free(host_data.addresses);
+		host_data.addresses = malloc(ctx->zs->elem_count * OT_IP6_ADDRESS_SIZE);
+	} else if (!host_data.addresses) {
+		host_data.addresses = malloc(ctx->zs->elem_count * OT_IP6_ADDRESS_SIZE);
+	}
+
+	host_data.addresses_num = ctx->zs->elem_count;
+
+	while (ctx->zs->elem_count) {
+		nrf_rpc_decode_buffer(ctx, &host_data.addresses[index++], OT_IP6_ADDRESS_SIZE);
+	}
+
+	zcbor_list_end_decode(ctx->zs);
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	error = otSrpClientSetHostAddresses(openthread_get_default_instance(), host_data.addresses,
+					    host_data.addresses_num);
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_uint(group, error);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_srp_client_set_host_addresses,
+			 OT_RPC_CMD_SRP_CLIENT_SET_HOST_ADDRESSES,
+			 ot_rpc_cmd_srp_client_set_host_addresses, NULL);
+
+static void ot_rpc_cmd_srp_client_is_auto_start_mode_enabled(const struct nrf_rpc_group *group,
+							     struct nrf_rpc_cbor_ctx *ctx,
+							     void *handler_data)
+{
+	bool enabled;
+
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	enabled = otSrpClientIsAutoStartModeEnabled(openthread_get_default_instance());
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_bool(group, enabled);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_srp_client_is_auto_start_mode_enabled,
+			 OT_RPC_CMD_SRP_CLIENT_IS_AUTO_START_MODE_ENABLED,
+			 ot_rpc_cmd_srp_client_is_auto_start_mode_enabled, NULL);
+
+
+
+static int ot_rpc_srp_client_init(void)
```

Comment:
This is not needed. `host_data` is in the static area so it will be zero-initialized.